### PR TITLE
Ensure properly synced before marking pilot ready

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/24117

Very WIP, need to clean this up a lot

### CODE REVIEW

File: pilot/pkg/model/push_context.go [hunk 1]
```diff
@@ -822,6 +824,7 @@ func (ps *PushContext) InitContext(env *Environment, oldPushContext *PushContext
 }
 
 func (ps *PushContext) createNewContext(env *Environment) error {
+	log.Errorf("howardjohn: create new context from %v", string(debug.Stack()))
```
Comment (costinm) [inline pilot/pkg/model/push_context.go:827]: @douglas-reid a "printf" like this is a sign this needs a varz...

File: pkg/queue/instance.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ type Instance interface {
 	Push(task Task)
 	// Run the loop until a signal on the channel
 	Run(<-chan struct{})
+	Size() int
```
Comment (costinm) [inline pkg/queue/instance.go:33]: I would really love to see 'queue' refactored and cleaned up... I thought it's no longer used with the event based interface - do you understand where/why is it still needed ?

Comment (howardjohn) [reply pkg/queue/instance.go:33]: Its used pretty much everywhere. Any events that come from k8s get put in the queue. That is how k8s implements their controllers aswell

**Comment by howardjohn** [comment]
not stale

**Pull request closed without merging**